pkg/controller/util/node/controller_utils.go

```diff
@@ -430,36 +357,23 @@ func GetTenantPartitionManagersFromKubeClients(clients []clientset.Interface, st
 }
 
 func GetTenantPartitionManagersFromServerNames(tenantServers []string, stop <-chan struct{}) ([]*TenantPartitionManager, error) {
-	clients, err := GetTenantPartitionClients(tenantServers)
+	clients, err := CreateTenantPartitionClients(tenantServers)
 	if err != nil {
 		return nil, err
 	}
 
 	return GetTenantPartitionManagersFromKubeClients(clients, stop)
 }
 
-// this check enforce the tenant-server option use the format [scheme];//[IP addrese]:port to specify the url
-func ValidateUrl(urlString string) error {
-	_, err := url.ParseRequestURI(urlString)
-	if err != nil {
-		return err
-	}
-
-	u, err := url.Parse(urlString)
-	if err != nil {
-		return err
-	}
-	if u.Scheme == "" {
-		return fmt.Errorf("Invalid url (%v): missing scheme", urlString)
-	}
-	if u.Host == "" {
-		return fmt.Errorf("Invalid url (%v): missing host", urlString)
-	}
-
-	host, _, _ := net.SplitHostPort(u.Host)
-	if net.ParseIP(host) == nil {
-		return fmt.Errorf("Invalid url (%v): invalid host IP", urlString)
+func CreateTenantPartitionClients(kubeconfigFiles []string) ([]clientset.Interface, error) {
```

Comment:
This function can also be moved to clientutil.go

Reply:
this is intentionally kept here. the refactored common functions in the client-go/util/clientutil.go ONLY intended to have functions to create single clientset. for array of them if needed, it belongs to the caller components logic.